src/main/java/core/basesyntax/FigureSupplier.java

```diff
@@ -0,0 +1,58 @@
+package core.basesyntax;
+
+import java.util.Random;
+
+public class FigureSupplier {
+    private static final int FIGURE_COUNT = 5;
+    private static final int RANDOM_BOUND = 30;
+    private static final int DEFAULT_CIRCLE_RADIUS = 10;
+    private Random random = new Random();
+    private ColorSupplier colorSupplier = new ColorSupplier();
+
+    public Figure getRandomFigure() {
+        int numberOfFigures = random.nextInt(FIGURE_COUNT);
+        Color randomColor = ColorSupplier.getRandomColor();
+        switch (numberOfFigures) {
+            case 0:
+                Circle circle = new Circle();
+                circle.setRadius(random.nextInt(RANDOM_BOUND));
+                circle.setColor(randomColor);
+                return circle;

```

Comment:
Better create private methods for creating all these figures (like getRandomCircle()) - it will be easier to read the code.